File: mafbase_stream/ios/Classes/Overlay/OverlayInvalidator.swift
import Foundation

/// Колбэк, который overlay-view дёргает, когда содержимое изменилось и нужно
/// пересобрать GL-текстуру оверлея.
///
/// Для статичных оверлеев — вызывается при явных изменениях. Для анимированных —
/// view сама подписывается на vsync (`CADisplayLink`) и вызывает `invalidate()`
/// в каждом tick.
@objc public protocol OverlayInvalidator: AnyObject {
    func invalidate()
}

File: mafbase_stream/android/src/main/kotlin/com/example/mafbase_stream/overlay/OverlayInvalidator.kt
package com.example.mafbase_stream.overlay

/**
 * Колбэк, который overlay-view дёргает, когда содержимое изменилось и нужно
 * пересобрать GL-текстуру оверлея.
 *
 * Для статичных оверлеев — вызывается при явных изменениях. Для анимированных —
 * view сама подписывается на vsync (`View.postInvalidateOnAnimation`,
 * `Choreographer.postFrameCallback`) и вызывает [invalidate] в каждом tick.
 */
fun interface OverlayInvalidator {
    fun invalidate()
}
